src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -375,6 +380,33 @@ public int compareOffsets(final NakadiCursor first, final NakadiCursor second) t
         return KafkaCursor.fromNakadiCursor(first).compareTo(KafkaCursor.fromNakadiCursor(second));
     }
 
+    public long totalEventsInPartition(final Timeline timeline, final String partitionString) {
+        final Timeline.StoragePosition positions = timeline.getLatestPosition();
+        if (positions == null) {
+            throw new InvalidCursorOperation(InvalidCursorOperation.Reason.PARTITION_NOT_FOUND);
```

Comment:
Why do you throw this kind of exception?
As I understand the situation that `timeline.getLatestPosition() == null` means that current timeline is still "active".